docs(sources): add integrations directory with 31 source pages (ETL-980) (#739)
glassflow/clickhouse-etl: docs/app/sources/postgresql/page.mdx

---
title: 'PostgreSQL'
description: 'Stream changes from PostgreSQL into ClickHouse with GlassFlow'
---
import { Callout } from 'nextra/components'

# PostgreSQL

PostgreSQL is an advanced open-source relational database known for reliability and extensibility. It supports complex queries, ACID transactions, and modern features like JSON and vector extensions.

## Using PostgreSQL with GlassFlow

GlassFlow can ingest changes from PostgreSQL into ClickHouse with the same pipeline capabilities available to all GlassFlow sources — deduplication, joins, filtering, and stateless transformations.

<Callout type="info">

> **Available on request.** PostgreSQL connectors are supported natively via the GlassFlow Enterprise Edition. To get started, [join our GlassFlow Slack community](https://glassflowhub.slack.com/join/shared_invite/zt-349m7lenp-IFeKSGfQwpJfIiQ7oyFFKg) to chat directly with our team, or [reach out](https://www.glassflow.dev/integrations#contact) to us with your use case, expected volumes, and timeline.

Reviewer: ```suggestion
**Available on request.** PostgreSQL connectors are natively supported in GlassFlow Enterprise Edition. To get started, [join our GlassFlow Slack community](https://glassflowhub.slack.com/join/shared_invite/zt-349m7lenp-IFeKSGfQwpJfIiQ7oyFFKg) to chat directly with our team, or [reach out](https://www.glassflow.dev/integrations#contact) to us with your use case, expected volumes, and timeline.
```
